### DISCUSSION TITLE
Manually set active workspace highlight in pop-up?

### DISCUSSION THREAD

**Discussion by humanplayer2** [mzur/gnome-shell-wsmatrix#162] [Q&A] [answered]
Dear all,

I have recently changed to the Nordic-darker theme on Pop!_OS 21.04 with COSMIC based on GNOME 3.38. 
All is good, except that I find the theme's color choice for the highlight of the active workspace in the pop-up too difficult to see (see image below)

I've tried poking around in the theme .css files, but can't seem to change the border color, making me think it is controlled somehow by wsmatrix.

I've tried poking around in the extension .js files, but don't know what I'm looking for.

Can I somehow manually set the highlight border color?

![The active workspace is that underneath the open browser.](https://user-images.githubusercontent.com/220154/125039913-51ae6800-e097-11eb-8485-5cb2446a6887.png)

**Comment by ebeem** [comment]
We tried to fix this back when #12 was reported, but this will need a large commit.
It's planned for `gnome 40` PR #152, but it's not implemented yet.
The old implementation used to get the workspace indicator, which is not very well styled by themes since `gnome-shell` doesn't show thumbnails. This makes a lot of styles of the popup switcher acceptable without thumbnails, but not very usable with them.

The `gnome 40` PR #152 implements the popup by inheriting the same popup as windows/applications/languages switcher. That one is more clean and better tested in styles but might not look as good as expected.

The plan for now is to try to have options for the appearance as mentioned in #12 as it will be impossible to make sure the `wsmatrix` looks as good as expected in all of `gnome-shell` themes. In case you're wondering, `wsmatrix` current implementation in both `gnome 40` and prior to `gnome 40` is 100% dependent on the shell theme.

**Reply by humanplayer2** [reply]
Thank you very much for the detailed answer. I've accepted @mzur's as it directly points to what I needed. EDIT: Recent updates broke that solution.

**Comment by mzur** [comment, marked as answer]
To add to @ebeem's answer, Workspace Matrix uses the `.workspace-thumbnail-indicator` class for the active workspace indicator. You should be able to find that in the theme CSS. If not, maybe you can use it to override the color.

**Reply by humanplayer2** [reply]
Thank you! This class is defined in the theme and makes exactly the change I was looking for!

**Reply by humanplayer2** [reply]
I have now updated to v30 and GNOME 40, and the `.workspace-thumbnail-indicator` class now only affects the indicator in the workspace overview, but not in the popup. I assume this is related to @ebeem 's answer, but I have alas been unable to operationalize that...

For v30 and GNOME 40, is there some other class I should look for if I want to change the active workspace indicator in the popup?

**Reply by mzur** [reply]
There isn't really a dedicated class for the active workspace indicator in the popup any more. You could try it with `.switcher-list.selected` but this will probably change the appearance of all switcher popups (e.g. the monitor selection popup), too. Please try and see if this works for you. Otherwise we could think about adding a dedicated class for custom styling.

**Reply by humanplayer2** [reply]
That class doesn't appear the used in the theme, but `.switcher-list .item-box:selected` is, wth both `color`  and `background-color` elements. Changing the `background-color` did the trick!

Thank you, again!
